lib/BackbeatMetadataProxy.js

```diff
@@ -105,6 +105,60 @@ class BackbeatMetadataProxy extends BackbeatTask {
         });
     }
 
+    headObject(params, log, cb) {
+        this.retry({
+            actionDesc: 'head object from source',
+            logFields: {
+                bucket: params.bucket,
+                objectKey: params.objectKey
+            },
+            actionFunc: done => this._headObjectOnce(params, log, done),
+            shouldRetryFunc: err => err.retryable,
+            log,
+        }, cb);
+    }
+
+    _headObjectOnce(params, log, cb) {
+        const { bucket, objectKey, locations } = params;
+        log.debug('heading object', {
+            where: 'source',
+            bucket, objectKey, locations,
+            method: 'BackbeatMetadataProxy._headObjectOnce',
+        });
+        const cbOnce = jsutil.once(cb);
+        const req = this.backbeatSource.multipleBackendHeadObject({
+            Bucket: bucket,
+            Key: objectKey,
+            Locations: JSON.stringify(locations.map(location => ({
```

Comment:
Need to follow-up to ensure this cannot exceed a max header size for the request.

Reply:
Limited to just a single array entity as this serves the purpose in the multiple backend of cloudserver.